### PR TITLE
PXP-8399 Update cryptography and migrate to poetry

### PR DESCRIPTION
**Pull request by MaribelleHGomez**
Modified .travis file for poetry use
Changed jwt auth_header test to account for change in return type

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
- Now uses poetry for package management

### Breaking Changes
pyJWT encode() now returns string instead of byte array

### Improvements
Migrated to poetry for package management

### Dependency updates
- pyJWT
- cryptography

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+version = "1.1.0"
+description = "This package includes several utility Python tools for the Gen3 stack."
+authors = ["CTDS UChicago <cdis@uchicago.edu>"]
+license = "Apache-2.0"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+cryptography = "^3.4.7"
+PyJWT = "^2.1.0"
+requests = "^2.25.1"
```
Comment (paulineribeyre) [inline pyproject.toml:12]: why was requests downgraded from `~=2.5` to `^2.25.1`?

File: .travis.yml [hunk 2]
```diff
@@ -3,29 +3,35 @@ dist: xenial
...
+- source $HOME/.poetry/env
+- which poetry
+- poetry install -vv
+- poetry show
+
+script:
+- poetry run pytest -vv ./test
+after_script:
+- pip install codacy-coverage
+- python-codacy-coverage -r coverage.xml
```
Comment (paulineribeyre) [inline .travis.yml:16]: codacy isn't set up in this repo so we don't need these

File: .travis.yml [hunk 3]
```diff
@@ -3,29 +3,35 @@ dist: xenial
...
-- if [ $(python setup.py --version) == '0.0.0' ]; then travis_terminate 1; fi
+- poetry config http-basic.pypi $PYPI_USER $PYPI_PASSWORD
+- poetry build
 deploy:
-  provider: pypi
-  user: uc-ctds
-  password:
-    secure: SgpCLTs5m220L/vzbntmPkm9DinEHV/hFG+fdXAobxwX237PnXIqeOBnAZClEOG8+yZRMa7NWyD1jGmuWPEQV4ptCYTIu8TRySUhSLvBe1hoghmJaUFrHng/HwdTItXFV5sSVvuIaQjg3Jq0QcjNjmo+T8sI1CD5w+83KiU16GDvyue9Lg1pbBV83YwDP2QqBhMlh6iVEKeVN9HFgu78r0BXloiQYKjVu66O6/X/JPXxQqyKsSencZPl7xzNBxKZC94QEGckK1l8V3RdGyu/CPeC6ubcew07A7ZgQZQy62zybvs/RW5YnyH4S9UYlVuSSzNmQtbWy+IIRYpujtxxsQ9c99dqVfRGf5szdXs31r9K5wHDbGqto1NSXKz1yjOVPVGgVGT/ZVNAUjAonH6KlUarc8URHVpvUoEzGEaf5e3I7Kp8tQWfM2fvabfG2tSonscaDpj2oybCBiJ5ds9FMLy74DJwy8YfIGvO8i0jXoWMVVbJUPzWGwgwdXN75LMUVF2peFsQXzJYDacSLLp+zwACjA7WuUNLH75SKLmXWtu1Bnc/ULHnMH/UxwKEXgPVrDaOZVvUNqrk1j8BlxaN6f17WGMvZW5YmRM/7TLcqRCweWsekt8vL8zp2Nx+Q2Wwadyssjardap03U+o8Tew4zDtgv4+tHeMpOfhs/99fpI=
+  provider: script
+  script: "poetry publish"
```
Comment (paulineribeyre) [inline .travis.yml:23]:
```suggestion
  script: poetry publish
```

File: .travis.yml [hunk 4]
```diff
@@ -3,29 +3,35 @@ dist: xenial
...
+- poetry run pytest -vv ./test
+after_script:
+- pip install codacy-coverage
+- python-codacy-coverage -r coverage.xml
+
 before_deploy:
-- sed -i.bak "s/=get_version()/='$TRAVIS_TAG'/g" setup.py
-- cat setup.py
-- if [ $(python setup.py --version) == '0.0.0' ]; then travis_terminate 1; fi
+- poetry config http-basic.pypi $PYPI_USER $PYPI_PASSWORD
```
Comment (paulineribeyre) [inline .travis.yml:19]: `PYPI_USER` and `PYPI_PASSWORD` need to be added to the encrypted travis env vars

Comment (MaribelleHGomez) [reply .travis.yml:15]: thanks

**Review by paulineribeyre — APPROVED** [review]
lgtm 😄

**Pull request merged**